src/fileformats/ocd_file_export.cpp

```diff
@@ -151,6 +151,68 @@ void copySymbolHead(const Symbol& source, Symbol& symbol)
 
 
 
+/**
+ * Test if the fill patterns constitute OCD structure "shifted rows".
+ */
+bool isStructureShiftedRows(const AreaSymbol::FillPattern& first, const AreaSymbol::FillPattern& second)
+{
+	auto renamed = [](PointSymbol const& symbol, QString const& name) {
+		auto clone = duplicate(symbol);
+		clone->setName(name);
+		return clone;
+	};
+	return first.type == AreaSymbol::FillPattern::PointPattern
+	       && second.type == AreaSymbol::FillPattern::PointPattern
+	       && first.rotatable() == second.rotatable()
+	       && qFuzzyCompare(first.angle, second.angle)
+	       && first.line_spacing == second.line_spacing
+	       && first.point_distance == second.point_distance
+	       && qAbs((first.line_offset - second.line_offset + first.line_spacing) % first.line_spacing - first.line_spacing / 2) <= 1
+	       && qAbs((first.offset_along_line - second.offset_along_line + first.point_distance) % first.point_distance - first.point_distance / 2) <= 1
+	       && first.point
+	       && second.point
+	       && first.point->equals(renamed(*second.point, first.point->getName()).get());
+}
+
+
+
+/**
+ * Postprocess a fill pattern for optimal OCD structure export.
+ */
+std::unique_ptr<PointSymbol> postProcessFillPattern(const AreaSymbol::FillPattern* first_point_pattern)
+{
+	auto point_symbol = duplicate(*first_point_pattern->point);
+	
+	// Move first pattern as needed
+	if (first_point_pattern->line_offset != 0
+	    || first_point_pattern->offset_along_line != 0)
```

Comment:
This would help for this particular commit, but not with the following one. 
What can be done is decomposition into more, smaller functions ("clean code").

Reply:
Ack.